src/xcode/ENA/ENA/Source/HTTPClient/Service/Services/_Service+Default.swift

```diff
@@ -69,18 +71,35 @@ extension Service {
 				}
 				
 				if let error = error {
-					completion(.failure(customError(in: resource, for: .transportationError(error))))
-					return
+					Log.info("No network connection (.transportationError)", log: .client)
+					// If we have no internet connection, we have to check first if we have a cached model for this resource.
+					if hasCachedData(resource) {
+						Log.info("Found some cached data", log: .client)
+						cached(resource, completion)
```

Comment:
I think we need this handling not just for the no network error but also for the other error scenarios (trust evaluation and server error codes 4xx/5xx)

Reply:
Is that really useful? The server responses an error - not found (404) or an internal server error and the client should response with a cached result? I'm no sure if that is the wanted behavior of an eTag based caching.

Reply:
Well, it is how the desired behaviour is defined in the [tech spec](https://github.com/corona-warn-app/cwa-app-tech-spec/blob/ce23af4fb75ff1ef0f42ef2ee284c11450ad8fbb/docs/spec/dcc-validation-service.md#requesting-the-validation-service-allowlist):

> If the request succeeds with status code 304 Not Modified, the operation shall return the most recent Validation Service Allowlist.
> 
> 
> If the request fails because of a missing or poor network connection, the operation shall log a corresponding error and return the most recent Validation Service Allowlist.
> 
> If the request fails with a client error (i.e. status code 40x), the operation shall log a corresponding error and return the most recent Validation Service Allowlist.
> 
> If the request fails with a server error (i.e. status code 50x), the operation shall log a corresponding error and return the most recent Validation Service Allowlist.

Reply:
You are right. For this we refactored a lot: 
- There is now a default value you can define in the resource, which will be returned instead of a fail.
- You can now define caching use cases for .noNetwork and for statusCode ranges per resource.
- And we added several tests for this.